packages/moleculer-db/moleculer-db.d.ts

```diff
@@ -1,22 +1,24 @@
 declare module "moleculer-db" {
   import { Context, ServiceBroker, Service } from "moleculer";
 
+  export interface DbServicePopulate extends Record<string,any>{
+     <R>(ids: string|number, docs: any[], rule: any, ctx: Context): Promise<R>;
+  }
   export interface DbServiceSettings {
     idField: string;
     fields?: Array<string>;
-    populates?: Array<any>;
+    populates?: Array<DbServicePopulate>;
     pageSize?: number;
     maxPageSize?: number;
     maxLimit?: number;
-    entityValidator?: any;
+    entityValidator?: <T, R>(entity: T)=> Promise<R>;
   }
-
-  export default class DbService<T=any,S extends DbServiceSettings = DbServiceSettings> extends Service<S> {
+ export interface DbEntity<T extends Record<string,any> = Record<string,any>> extends Record<string,any> {}
+  export default class DbService<T,S extends DbServiceSettings = DbServiceSettings> extends Service<S> {
  
   }
 
-  export interface QueryOptions<T> {
-    [name: string]: any;
+  export interface QueryOptions<T  = any> extends Record<string,any> {
```

Comment:
I recognize that this precedes you, but the generic `T` doesn't appear to be used.  That pattern seems to be very prevalent in this declaration file with many generics not being used.

@icebob Since I haven't looked at these types before, do you know if there was a rationale to adding these unused generics?

Reply:
I don't know. This d.ts has been created by @saeedtabrizi  here: https://github.com/moleculerjs/moleculer-db/commit/1680d56a4b7251030161fa498d873993c4183740